stonesoup/models/base.py

```diff
@@ -196,7 +196,7 @@ class GaussianModel(Model):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.random_state = np.random.RandomState(self.seed)
+        self.random_state = np.random.RandomState(self.seed) if self.seed is not None else None
```

Comment:
```suggestion
        self.random_state = np.random.RandomState(self.seed) if self.seed else\
            np.random.RandomState(np.random.get_state()[1])
```
Ah, ok, I understand the problem now.  I didn't realise the default random state used a new seed. 

The problem we have is that we should really be doing this stuff with default_rng, as numpy random seed is now legacy code.
https://albertcthomas.github.io/good-practices-random-number-generators/

Unfortunately, this is not in the numpy version we have at dstl (maybe this will change soon?) but for now we need to use seeds/random states. 

Numpy has the get_state method for returning the global state. For now we can pass this as the seed and I think this fixes the reproducibility problem and keeps self.random_state as a random_state. 

The documentation for get_state explicitly says this should not be necessary, but it seems that best practice is to pass the RNG/random state around if you want reproducible results, so any method to use a default global state seems to be a bit of a hack (see [here](https://github.com/numpy/numpy/issues/15322)). I agree this is kind of a pain, but I do get the reasons behind it.

Anyhow, I think this is the least bad way to do things for now, until we get the latest numpy as dstl. At that point we should probably completely move away from random seeds, use default_rng, and get into the habit of passing it around. 

That or I've misunderstood something and really overcomplicated this (which is always a possibility).

Reply:
I've open a new issue #462 to implement a more permanent solution. We should be getting a newer version of Numpy at Dstl soon which will allow us to use the current best practice. In the mean time, either solution works for the time being, and as it is only a temporary fix I'm not that fussed. 

Thank you @joldf for bringing this to our attention. I didn't realise my new way of producing reproducible random states had meant global random states stopped working.